Lib/http/client.py

```diff
@@ -1079,6 +1088,9 @@ def putrequest(self, method, url, skip_host=False,
         self._method = method
         if not url:
             url = '/'
+        # Prevent CVE-2019-9740.
+        if _contains_disallowed_url_pchar_re.search(url):
+            raise ValueError(f"URL can't contain control characters. {url!r}")
```

Comment:
Would it be possible to mention the first invalid character to help users to debug their code?